test/lib/testserver.js

```diff
@@ -50,7 +63,14 @@ TestServer.start = function () {
     init();
   }
   do {
-    server = app.listen(port);
+    server = app.listen(port)
+      .on('error', function(err) {
+        if ('EADDRINUSE' === err.code) {
+          // Just continue trying.
```

Comment:
Fair concern, but it isn't incompatible:

```js
// We keep trying on errors only if because port already in use
if ('EADDRINUSE' !== err.code)
```
(or better wording)

Having an empty `if` block is really odd...